modules/dynunet_pipeline/create_network.py

```diff
@@ -7,22 +7,38 @@
 
 
 def get_kernels_strides(task_id):
+    """
+    This function is only used for decathlon datasets with the provided patch sizes.
+    When refering this method for other tasks, please ensure that the patch size for each spatial dimension should
+    be divisible by the product of all strides in the corresponding dimension.
+    In addition, the minimal spatial size should have at least one dimension that has twice the size of
+    the product of all strides. For patch sizes that cannot find suitable strides, an error will be raised.
+
+    """
     sizes, spacings = patch_size[task_id], spacing[task_id]
+    input_size = sizes
     strides, kernels = [], []
-
+    min_patch_size = [1, 1, 1] * len(sizes)
     while True:
         spacing_ratio = [sp / min(spacings) for sp in spacings]
         stride = [
             2 if ratio <= 2 and size >= 8 else 1
             for (ratio, size) in zip(spacing_ratio, sizes)
         ]
+        min_patch_size = [i * j for i, j in zip(min_patch_size, stride)]
         kernel = [3 if ratio <= 2 else 1 for ratio in spacing_ratio]
         if all(s == 1 for s in stride):
             break
+        for idx, (i, j) in enumerate(zip(sizes, stride)):
+            if i % j != 0:
+                raise ValueError(
+                    f"patch size error, please try to modify the size {input_size[idx]} in the spatial dimension {idx}."
```

Comment:
This would be better if it said something specific about the condition being checked that helped the user choose better values. I'm not totally sure that's the condition that's being checked here but it would help more to say something about what is checked and how to fix it, perhaps something like f"Patch size error, please try to modify the size {input_size[idx]} in the spatial dimension {idx} to be an even multiple of the patch size."

Reply:
You updated the code as I wrote this! The new message looks better, I suppose it's too complex to say anything more specific about what patch size is supported.